velox/functions/prestosql/DateTimeFunctions.h

```diff
@@ -1394,4 +1395,35 @@ struct TimeZoneMinuteFunction : public TimestampWithTimezoneSupport<T> {
   }
 };
 
+template <typename T>
+struct ToISO8601FromTimestampWithTimezoneFunction : public TimestampWithTimezoneSupport<T> {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<Varchar>& result,
+      const arg_type<TimestampWithTimezone>& timestampWithTimezone) {
+    const auto milliseconds = *timestampWithTimezone.template at<0>();
+    Timestamp timestamp = Timestamp::fromMillis(milliseconds);
+    auto tsStr = timestamp.toString().substr(0, 23);
+
+    auto timezone =
+        util::getTimeZoneName(*timestampWithTimezone.template at<1>());
+
+    // ISO8601 output should contain timezone as UTC offset.
+    if (isalpha(timezone[0])) {
```

Comment:
I believe the date library in Velox has an API for this.

Reply:
Wow, I really reinvented the wheel here.. should've looked deeper into the existing APIs first. Replaced a lot of that with an invocation of `toTimezone()` - though the call to `date::make_zoned()` is still necessary in the case of a named timezone, to derive the appropriate UTC offset for the ISO string.